File: android/LearnJNI/app/src/main/java/com/example/learnjni/MainActivity.java
package com.example.learnjni;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.example.learnjni.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity
{

	// Used to load the 'learnjni' library on application startup.
	static
	{
		System.loadLibrary("learnjni");
	}

	private ActivityMainBinding binding;

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		binding = ActivityMainBinding.inflate(getLayoutInflater());
		setContentView(binding.getRoot());

		// Example of a call to a native method
		TextView tv = binding.sampleText;
		tv.setText(LearnJNI.getInstance().stringFromJNI());
	}


}

File: android/LearnLinphone/app/src/main/java/com/georgebindragon/learn/linphone/MainActivity.kt
package com.georgebindragon.learn.linphone

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import org.linphone.core.Factory

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun init(view: View) {

        // Core is the main object of the SDK. You can't do much without it.
        // To create a Core, we need the instance of the Factory.
        val factory = Factory.instance()

        // Some configuration can be done before the Core is created, for example enable debug logs.
        factory.setDebugMode(true, "Hello Linphone")

        // Your Core can use up to 2 configuration files, but that isn't mandatory.
        // On Android the Core needs to have the application context to work.
        // If you don't, the following method call will crash.
        val core = factory.createCore(null, null, this)

        // Now we can start using the Core object
        findViewById<TextView>(R.id.core_version).text = core.version
    }
}